modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManagerTest.java

```diff
@@ -0,0 +1,881 @@
+                    .thenReturn(getEncodeSCIMExceptionObject(new NotImplementedException()));
+
+            SCIMResponse scimResponse = groupResourceManager.delete(id, userManager);
+            Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+        } else if (expectedScimResponseStatus == ResponseCodeConstants.CODE_BAD_REQUEST) {
+
+            doThrow(new BadRequestException()).when(userManager).deleteGroup(id);
+
+            when(AbstractResourceManager.encodeSCIMException(any(BadRequestException.class)))
+                    .thenReturn(getEncodeSCIMExceptionObject(new BadRequestException()));
+
+            SCIMResponse scimResponse = groupResourceManager.delete(id, userManager);
+            Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+        }
+    }
+
+    @Test
+    public void testDeleteGroupCharonExceptionOnly()
+            throws NotFoundException, NotImplementedException, BadRequestException,
+            CharonException, InternalErrorException {
+
+        Group group = getNewGroup();
+        String id = group.getId();
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_GROUP_ENDPOINT + "/" + GROUP_ID);
+        doThrow(new CharonException()).when(userManager).deleteGroup(id);
+        when(AbstractResourceManager.encodeSCIMException(any(CharonException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new CharonException()));
+        SCIMResponse scimResponse = groupResourceManager.delete(id, userManager);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_INTERNAL_ERROR);
+    }
+
+    @DataProvider(name = "dataForTestUpdateGroupWithPUTSuccess")
+    public Object[][] dataToTestUpdateGroupWithPUTSuccess()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+
+        Group groupOld = decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING, schema, new Group());
+        String id = groupOld.getId();
+
+        Group groupNew = decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING_UPDATED, schema, new Group());
+
+        return new Object[][]{
+                {id, NEW_GROUP_SCIM_OBJECT_STRING_UPDATED, "id", null, groupNew, groupOld}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestUpdateGroupWithPUTSuccess")
+    public void testUpdateGroupWithPUTSuccess(String id, String scimObjectString, String
+            attributes, String excludeAttributes, Object objectNEWGroup, Object objectOLDGroup)
+            throws BadRequestException, NotFoundException, CharonException, NotImplementedException {
+
+        Group groupNew = (Group) objectNEWGroup;
```

Comment:
casting is not required if we directly pass the param with type as `Group` instead of passing it as an `Object`.